### PR TITLE
(SUP-3688) Plans to provision and import metrics

### PR DESCRIPTION
**Pull request by m0dular**
This commit adds two Puppet Plans to provision a dashboard node and
import archive metrics from a PE support script.

### CODE REVIEW

**Review by jarretlavallee — CHANGES_REQUESTED** [review]
Looks good to me. I added a couple of nit picks and one potential problem.

File: plans/load_metrics.pp [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      ensure  => file,
+      content => Deferred('inline_epp', [file('puppet_operational_dashboards/telegraf.conf'), $telegraf_vars]),
+    }
+    file { "${conf_dir}/telegraf.conf.d":
+      ensure => directory,
+    }
+    ['postgres.conf', 'puppetdb.conf', 'puppetserver.conf', 'system_cpu.conf', 'system_memory.conf', 'system_procs.conf'].each |$script| {
+      file { "${conf_dir}/telegraf.conf.d/${script}":
+        ensure => file,
+        source => "puppet:///modules/puppet_operational_dashboards/scripts/${script}",
```
[the remark this answers is not in the document]
Comment (m0dular) [reply plans/load_metrics.pp:105]: I'm honestly not sure to call these scripts or config files, they're kind of both.  They do have code that gets executed, but they're also parameters to the `--config-directory` flag.  What directory do you think they should go in?

File: plans/load_metrics.pp [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      [file('puppet_operational_dashboards/datasource.epp'), $token_vars]),
+      notify  => Service['grafana-server'],
+    }
+
+    $telegraf_vars = {
+      bucket => $influxdb_bucket,
+      org => $influxdb_org,
+      port => $influxdb_port,
+      host => $targets,
+      token => Sensitive(Deferred('influxdb::retrieve_token', ["http://${targets}:8086", $telegraf_token, $token_file])),
```
[the remark this answers is not in the document]
Comment (m0dular) [reply plans/load_metrics.pp:89]: Should we loop over each target, or enforce that the plan only accepts a single one?

File: plans/load_metrics.pp [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  String $influxdb_org = 'puppetlabs',
+  String $influxdb_bucket = 'puppet_data',
+  Integer $influxdb_port = 8086,
+  String $grafana_datasource = $influxdb_bucket,
+  String $telegraf_token = 'puppet telegraf token',
+  String $token_file = '/root/.influxdb_token',
+  String $conf_dir = '/tmp/telegraf',
+  #TODO
+  Enum['local', 'remote'] $telegraf_process = 'remote',
+  String $metrics_dir,
```
[the remark this answers is not in the document]
Comment (m0dular) [reply plans/load_metrics.pp:42]: Agreed, I feel like this should be a convention enforced by PDK?

File: plans/load_metrics.pp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jarretlavallee) [reply plans/load_metrics.pp:105]: At the end of the day, it doesn't really matter. How about neither and go with something that denotes that these are files that the plans use or denote the purpose? `archive`, `plan_files` or something better

File: plans/load_metrics.pp (hunk 2 above)
[the remark this answers is not in the document]
Comment (jarretlavallee) [reply plans/load_metrics.pp:89]: I can't see a reason that target would be more than one. Is there a use case for that? Maybe we just force it to be one and error out if there is more than one. At that point, we would have a use case to make a change to it.

[the remark this answers is not in the document]
Comment (m0dular) [reply plans/load_metrics.pp:89]: I could _maybe_ see wanting to ship metrics from one archive to multiple InfluxDB hosts, but that's probably very fringe.  We can enforce only a single target, at least for now.

**Review by jarretlavallee — APPROVED** [review]

**Pull request merged**